geth/api/backend_test.go

```diff
@@ -209,7 +209,7 @@ func (s *BackendTestSuite) TestCallRPC() {
 		{
 			`{"jsonrpc":"2.0","method":"shh_version","params":[],"id":67}`,
 			func(resultJSON string) {
-				expected := `{"jsonrpc":"2.0","id":67,"result":"5.0"}` + "\n"
+				expected := `{"jsonrpc":"2.0","result":"5.0"}`
```

Comment:
@adambabik good question. The only user of this response is status-react, so I talked to react team asking exactly that - if they use this `id` in any way or not. Response was negative, they only care about `result` field. 

This `id` field is indeed used internally by go-ethereum's RPC client we're using here. This client returns only `result` to us, so there is no way to know the `id`. And, assuming we don't need it for our needs, that's fine.

Reply:
Ok, that makes sense if they don't use it at all. I guess it's not JSON RPC anymore when status-react talk to us but some custom bindings.

Reply:
@adambabik unfortunately it still is. So we have JSON-RPC format serializing/deserializing loginc both in status-react and status-go. Actually, it's not Clojure code, but web3.js itself creates JSON-RPC requests.

Reply:
@divan Let's clarify that. 

If status-react uses `CallRPC()` binding, status-go receives this call, parses it and executes via RPC client in upstream or local node. The RPC server responds and we send back only `result` property. Is it correct?

If status-react executes a code in jail, web3.js makes a call to the RPC server, which I guess is in status-go. Next, the RPC server decides where to forward this call -- local or upstream node. The response should be a valid JSON-RPC response though. Is it right?

Reply:
@adambabik that's correct. `CallRPC()` sends raw JSON and expects raw JSON as response. When underlying RPC client executes call and receives JSON-RPC reply it parses it, takes the `result` property, unmarshal it into object and returns to our RPC client. Our client then constructs JSON-RPC response manually in `makeSuccessResponse` (https://github.com/status-im/status-go/pull/311/files/c47653efe399fcff7bcd72cbed29d1e51d1a59d1#diff-670fcb460fc5e655da2af8083cba7ee2R105).

It works because, according to spec, successful JSON-RPC responses basically contain only three fields:
```{
  "id":67,
  "jsonrpc":"2.0",
  "result": "some result"
}
```

`jsonrpc` is constant, result we're constructing from that object, and `id` was the field in question. As clojure devs confirmed they not relying on it in any case, so we're fine with sending back to web3.js it without `id`:

```
{
  "jsonrpc":"2.0",
  "result": "some result"
}
```

Reply:
Ok, great. So my only concern is that web3.js may use `id` for something as it's an external lib and it can actually expect valid JSON-RPC response.

BTW. I found that we do something like that:
```js
var web3 = new Web3(ourProvider);
```

And `ourProvider` is this `jeth` from https://github.com/status-im/status-go/blob/develop/geth/jail/jail.go#L105 and it has `send` method explicitly set to `jail.Send`. Now it's clear how we intercept `web3.eth.*` calls!

Reply:
@adambabik I just checked the `web3.js` code and that maybe an issue. :/

@rasom does this line in `web3.js` mean that JSON-RPC response without `id` will be discarded?
https://github.com/ethereum/web3.js/blob/develop/lib/web3/jsonrpc.js#L66

As I understood the logic of web3.js here, it will be enough to add any `id` (0 for example). Would be nice to have e2e tests for that to catch problem in case if web3.js will change its logic in the future.

Reply:
@adambabik just added default ID (0) for responses! Should be sufficient to bypass web3.js validation.

Reply:
I would strongly advised to just add that `id` and make it a valid JSON-RPC request/response :P 

@divan in `jail_test.go` there are tests which uses web3.js. Check out e.g. `TestJailRPCSend`.

Reply:
@divan wouldn't it be a better strategy to check if `id` is in the request and if so pass it back in the response? That's how `id` is supposed to work in JSON-RPC. Otherwise, we can fallback to a default.

Reply:
@adambabik great idea!